COSMIC Beta

HeitorAugustoLN opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Preparing new packages and package updates for COSMIC Desktop Environment Beta
cc: @NixOS/cosmic

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

HeitorAugustoLN committed:
cosmic-initial-setup: init at 1.0.0-beta.1.1

HeitorAugustoLN committed:
nixos/cosmic: add cosmic-initial-setup

a-kenji commented on pkgs/by-name/co/cosmic-initial-setup/package.nix:
```diff
@@ -0,0 +1,86 @@
+  ];
+
+  postPatch = ''
+    substituteInPlace justfile \
```
Our current contributing guidelines encourage you to also explain why you used `substituteInPlace`:

>   - If any (especially opinionated) patch or `substituteInPlace` is applied, document why.

HeitorAugustoLN commented:
> Are there any other packages that also need to be packaged for the beta?

No, the only unpackaged COSMIC software is [cosmic-osk](https://github.com/pop-os/cosmic-osk), but I am pretty sure this package is not nearly ready for use yet

HeitorAugustoLN committed:
cosmic-icons: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-icons/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-randr: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-randr/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-settings: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-settings/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-settings-daemon: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-settings-daemon/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-settings-daemon: remove `with lib` from meta

HeitorAugustoLN committed:
cosmic-comp: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-comp/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-applets: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-applets/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-launcher: 1.0.0-alpha.7 -> 1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-applibrary: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-applibrary/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-files: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-files/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-panel: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-panel/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-notifications: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-notifications/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-idle: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-idle/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-workspaces-epoch: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-workspaces-epoch/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-greeter: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-greeter/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
nixos/cosmic-greeter: update default session

HeitorAugustoLN committed:
cosmic-osd: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-osd/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
xdg-desktop-portal-cosmic: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/xdg-desktop-portal-cosmic/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-bg: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-bg/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-wallpapers: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-wallpapers/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-session: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-session/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-screenshot: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-screenshot/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-edit: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-edit/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-edit: use libcosmicAppHook

HeitorAugustoLN committed:
cosmic-edit: disable just check

Makes sure that we run the standard rust package
checkPhase

HeitorAugustoLN committed:
cosmic-edit: remove `with lib` from meta

HeitorAugustoLN committed:
cosmic-edit: add updateScript

HeitorAugustoLN committed:
cosmic-player: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-player/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-store: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-store/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN committed:
cosmic-term: 1.0.0-alpha.7 -> 1.0.0-beta.1

Diff: https://github.com/pop-os/cosmic-term/compare/refs/tags/epoch-1.0.0-alpha.7...epoch-1.0.0-beta.1

HeitorAugustoLN commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 440950`
Commit: `2a20a803e9a60b7e34c94ccaf86ad4b87ae7fbf5`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:x: 24 packages failed to build:</summary>
  <ul>
    <li>cosmic-applets</li>
    <li>cosmic-applibrary</li>
    <li>cosmic-bg</li>
    <li>cosmic-comp</li>
    <li>cosmic-comp.debug</li>
    <li>cosmic-edit</li>
    <li>cosmic-files</li>
    <li>cosmic-greeter</li>
    <li>cosmic-idle</li>
    <li>cosmic-launcher</li>
    <li>cosmic-notifications</li>
    <li>cosmic-osd</li>
    <li>cosmic-panel</li>
    <li>cosmic-player</li>
    <li>cosmic-reader</li>
    <li>cosmic-session</li>
    <li>cosmic-store</li>
    <li>cosmic-term</li>
    <li>cosmic-workspaces-epoch</li>
    <li>cosmic-workspaces-epoch.debug</li>
    <li>examine</li>
    <li>forecast</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 14 packages built:</summary>
  <ul>
    <li>cosmic-design-demo</li>
    <li>cosmic-ext-calculator</li>
    <li>cosmic-ext-tweaks</li>
    <li>cosmic-icons</li>
    <li>cosmic-initial-setup</li>
    <li>cosmic-randr</li>
    <li>cosmic-screenshot</li>
    <li>cosmic-settings</li>
    <li>cosmic-settings-daemon</li>
    <li>cosmic-wallpapers</li>
    <li>libcosmicAppHook</li>
    <li>oboete</li>
    <li>quick-webapps</li>
    <li>tasks</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:x: 36 packages failed to build:</summary>
  <ul>
    <li>cosmic-applets</li>
    <li>cosmic-applibrary</li>
    <li>cosmic-bg</li>
    <li>cosmic-comp</li>
    <li>cosmic-comp.debug</li>
    <li>cosmic-design-demo</li>
    <li>cosmic-edit</li>
    <li>cosmic-ext-calculator</li>
    <li>cosmic-ext-tweaks</li>
    <li>cosmic-files</li>
    <li>cosmic-greeter</li>
    <li>cosmic-idle</li>
    <li>cosmic-initial-setup</li>
    <li>cosmic-launcher</li>
    <li>cosmic-notifications</li>
    <li>cosmic-osd</li>
    <li>cosmic-panel</li>
    <li>cosmic-player</li>
    <li>cosmic-randr</li>
    <li>cosmic-reader</li>
    <li>cosmic-screenshot</li>
    <li>cosmic-session</li>
    <li>cosmic-settings</li>
    <li>cosmic-settings-daemon</li>
    <li>cosmic-store</li>
    <li>cosmic-term</li>
    <li>cosmic-workspaces-epoch</li>
    <li>cosmic-workspaces-epoch.debug</li>
    <li>examine</li>
    <li>forecast</li>
    <li>libcosmicAppHook</li>
    <li>oboete</li>
    <li>quick-webapps</li>
    <li>tasks</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>cosmic-icons</li>
    <li>cosmic-wallpapers</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cosmic-wallpapers</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cosmic-wallpapers</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>cosmic-applets</summary>
<pre>137 |     pub fn popup_view(&amp;self) -&gt; cosmic::Element&lt;&#x27;_, Msg&gt; {
    |                                                 +++

warning: hiding a lifetime that&#x27;s elided elsewhere is confusing
   --&gt; cosmic-applet-status-area/src/components/status_menu.rs:168:24
    |
168 | fn layout_view(layout: &amp;Layout, expanded: Option&lt;i32&gt;) -&gt; cosmic::Element&lt;Msg&gt; {
    |                        ^^^^^^^                            -------------------- the same lifetime is hidden here
    |                        |
    |                        the lifetime is elided here
    |
    = help: the same lifetime is referred to in inconsistent ways, making the signature confusing
help: use `&#x27;_` for type paths
    |
168 | fn layout_view(layout: &amp;Layout, expanded: Option&lt;i32&gt;) -&gt; cosmic::Element&lt;&#x27;_, Msg&gt; {
    |                                                                           +++

warning: `cosmic-applet-minimize` (lib) generated 3 warnings
warning: `cosmic-applet-status-area` (lib) generated 4 warnings
   Compiling cosmic-applets v0.1.1 (/build/source/cosmic-applets)</pre>
</details>
<details>
<summary>cosmic-applibrary</summary>
<pre>     = note: `#[warn(mismatched_lifetime_syntaxes)]` on by default
help: use `&#x27;_` for type paths
     |
1054 |     fn view(&amp;self) -&gt; Element&lt;&#x27;_, Message&gt; {
     |                               +++

warning: hiding a lifetime that&#x27;s elided elsewhere is confusing
    --&gt; src/app.rs:1058:20
     |
1058 |     fn view_window(&amp;self, id: SurfaceId) -&gt; Element&lt;Message&gt; {
     |                    ^^^^^                    ---------------- the same lifetime is hidden here
     |                    |
     |                    the lifetime is elided here
     |
     = help: the same lifetime is referred to in inconsistent ways, making the signature confusing
help: use `&#x27;_` for type paths
     |
1058 |     fn view_window(&amp;self, id: SurfaceId) -&gt; Element&lt;&#x27;_, Message&gt; {
     |                                                     +++
</pre>
</details>
<details>
<summary>cosmic-bg</summary>
<pre>   Compiling xdg v2.5.2
   Compiling cosmic-config v0.1.0 (https://github.com/pop-os/libcosmic#364c0b93)
   Compiling document-features v0.2.11
   Compiling rand_chacha v0.9.0
   Compiling backtrace v0.3.75
   Compiling jxl-render v0.12.3
   Compiling jxl-jbr v0.2.1
   Compiling derive_setters v0.1.8
   Compiling wayland-protocols-misc v0.3.9
   Compiling wayland-protocols-experimental v20250721.0.1
   Compiling calloop-wayland-source v0.4.0
   Compiling csscolorparser v0.7.2
   Compiling xkbcommon v0.8.0
   Compiling jxl-oxide v0.12.4
   Compiling colorgrad v0.7.2
   Compiling cosmic-bg-config v0.1.0 (/build/source/config)
   Compiling color-eyre v0.6.5
   Compiling rand v0.9.2
   Compiling fast_image_resize v5.1.4
   Compiling cosmic-bg v0.1.0 (/build/source)</pre>
</details>
<details>
<summary>cosmic-comp</summary>
<pre>@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;updateAutotoolsGnuConfigScriptsPhase&quot; }
Running phase: configurePhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;configurePhase&quot; }
Running phase: buildPhase
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;buildPhase&quot; }
Executing cargoBuildHook
cargoBuildHook flags: -j 6 --target x86_64-unknown-linux-gnu --offline --profile release
   Compiling proc-macro2 v1.0.97
   Compiling unicode-ident v1.0.18
   Compiling cfg-if v1.0.1
   Compiling libc v0.2.175
   Compiling serde v1.0.219
   Compiling log v0.4.27
   Compiling pkg-config v0.3.32
   Compiling once_cell v1.21.3
   Compiling smallvec v1.15.1
   Compiling version_check v0.9.5
   Compiling rustix v1.0.8
   Compiling autocfg v1.5.0
   Compiling linux-raw-sys v0.9.4</pre>
</details>
<details>
<summary>xdg-desktop-portal-cosmic</summary>
<pre>247 |     pub app_id: String,
    |         ^^^^^^
248 |     pub parent_window: String,
    |         ^^^^^^^^^^^^^
    |
    = note: `Args` has derived impls for the traits `Debug` and `Clone`, but these are intentionally ignored during dead code analysis
    = note: `#[warn(dead_code)]` on by default

warning: methods `feedback_formats` and `modifiers_for_format` are never used
  --&gt; src/wayland/mod.rs:71:8
   |
64 | impl DmabufHelper {
   | ----------------- methods in this implementation
...
71 |     fn feedback_formats(&amp;self) -&gt; impl Iterator&lt;Item = &amp;DmabufFormat&gt; {
   |        ^^^^^^^^^^^^^^^^
...
79 |     pub fn modifiers_for_format(&amp;self, format: u32) -&gt; impl Iterator&lt;Item = u64&gt; + &#x27;_ {
   |            ^^^^^^^^^^^^^^^^^^^^
</pre>
</details>
</details>

thefossguy commented:
@Pandapip1 Could you specify if it is okay to drop your patch? Especially since your PR was closed _by you_.
https://github.com/NixOS/nixpkgs/blob/489ccd84df4303db7f80b200c67504a7945e4ded/pkgs/by-name/co/cosmic-wallpapers/package.nix#L24

(I also raised this during Alpha 7 but received no reply. https://github.com/NixOS/nixpkgs/pull/401497#issuecomment-2828794431)

Pandapip1 commented:
Sure thing. At this point I'm not using it on KDE Plasma or GNOME anymore.

thefossguy commented on pkgs/by-name/co/cosmic-applets/package.nix:
```diff
@@ -20,38 +21,32 @@
 
-  cargoHash = "sha256-wWs3B5hh2DP93i+4gGDTi+7NT4bj8ULJ+fT95sXxUdg=";
-
-  patches = [
-    (fetchpatch2 {
-      name = "fix-bluetooth-dbus-spam.patch";
-      url = "https://github.com/pop-os/cosmic-applets/commit/b6bb982f2dace0a3d19c78b4b4247760a8010d5b.patch?full_index=1";
-      hash = "sha256-S5F9rqYrB38T9R6i/n/j3s79Xeh6BMmNkC+E2kTsus4=";
-    })
-  ];
```
This removal should also include the removal of `fetchpatch2` from the package's inputs.

thefossguy commented on pkgs/by-name/co/cosmic-edit/package.nix:
```diff
@@ -3,40 +3,31 @@
 
-  cargoHash = "sha256-qfPLDgGeYGSO0ZKJooXRK0NnTqzJ6zq6RhBpTTUusQY=";
-
-  # COSMIC applications now uses vergen for the About page
-  # Update the COMMIT_DATE to match when the commit was made
-  env.VERGEN_GIT_COMMIT_DATE = "2025-04-17";
-  env.VERGEN_GIT_SHA = finalAttrs.src.tag;
```
While the `libcosmic-app-hook.sh` script does set `VERGEN_GIT_COMMIT_DATE`, I don't think that setting `VERGEN_GIT_SHA` is automated yet. Is it or am I being an idiot?

HeitorAugustoLN commented on pkgs/by-name/co/cosmic-files/package.nix:
```diff
@@ -12,26 +12,22 @@
 
-  cargoHash = "sha256-7AOdSk9XIXFCDyCus3XgOK3ZBVa4CvX+NFM0jHf7Wbs=";
-
-  env = {
-    VERGEN_GIT_COMMIT_DATE = "2025-04-22";
-    VERGEN_GIT_SHA = finalAttrs.src.tag;
-  };
```
`cosmic-files` doesn't use vergen anymore too

thefossguy commented:
@HeitorAugustoLN I've noticed that there is no need to exclude _any_ tests (yay!) in the `cosmic-files` package anymore. Could you please also remove that block **after testing it on your end too**? Thank you in advance!

HeitorAugustoLN commented on pkgs/by-name/co/cosmic-osd/package.nix:
```diff
@@ -5,37 +5,44 @@
+    sound-theme-freedesktop
   ];
 
+  preFixup = ''
+    libcosmicAppWrapperArgs+=(--suffix XDG_DATA_DIRS : "${sound-theme-freedesktop}/share")
+  '';
+
```
I quite like the idea

HeitorAugustoLN committed:
nixos/cosmic: enable xdg sounds

thefossguy commented on pkgs/by-name/co/cosmic-files/package.nix:
```diff
@@ -72,17 +68,7 @@ rustPlatform.buildRustPackage (finalAttrs: {
-    # is disabled in the Nix sandbox and the tests fail because they can't
-    # run in the sandbox. Ideally, the `compio` crate should fallback to a
-    # non-io_uring runtime but for some reason, that doesn't happen.
-    cargoTestFlags="$defaultCargoTestFlags --package cosmic-files -- \
```
The line `cargoTestFlags="$defaultCargoTestFlags --package cosmic-files` was removed. This means that _both_ apps' tests (for `cosmic-files` and `cosmic-files-applet`) run here, instead of separately. Since there's no real need to run both apps' tests separately, should we just consolidate them by removing the explicit specification of running tests for `cosmic-files-applet`?

HeitorAugustoLN replied:
It wasn't removed. About making them into one, I am not sure about it for now

thefossguy replied:
I read the diff wrong, sorry!

HeitorAugustoLN commented on pkgs/by-name/co/cosmic-files/package.nix:
```diff
@@ -12,26 +12,22 @@
   nativeBuildInputs = [
     just
     libcosmicAppHook
+    rustPlatform.bindgenHook
```
Yes, `cosmic-files` now has the `bindgen` dependency.

https://github.com/pop-os/cosmic-files/commit/3beda45a6f81d96cbc054f45c2b168b1f9e111b9

thefossguy commented:
@HeitorAugustoLN I think `VERGEN_GIT_COMMIT_DATE` and `VERGEN_GIT_SHA` need to be removed from the `cosmic-greeter` package's definition too. :)

HeitorAugustoLN commented:
> @HeitorAugustoLN I think `VERGEN_GIT_COMMIT_DATE` and `VERGEN_GIT_SHA` need to be removed from the `cosmic-greeter` package's definition too. :)

It is still being used, but I forgot to update the commit date

thefossguy commented:
@HeitorAugustoLN the NixOS 25.11 release reminds me that we should also mention that the 'Beta 1' is now included in the nixpkgs repository, also nothing that it was upgraded from 'Alpha 7'.

HeitorAugustoLN commented:
> And if you do `nix build .#cosmic-inital-setup --store /tmp/test` does it work?

No, it does not, lol

HeitorAugustoLN commented on pkgs/by-name/co/cosmic-applets/package.nix:
```diff
@@ -20,38 +20,32 @@
     glib
     libinput
     pulseaudio
+    pipewire
```
They were added as dependencies for the Pop!_OS package, so probably yes.

HeitorAugustoLN commented on pkgs/by-name/co/cosmic-initial-setup/package.nix:
```diff
@@ -0,0 +1,93 @@
+
+  # cargo-auditable fails during the build when compiling the `crabtime::function`
+  # procedural macro. It panics because the `--out-dir` flag is not passed to
+  # the rustc wrapper.
```
I haven't investigated this issue, but it happens in this function:

https://github.com/pop-os/cosmic-initial-setup/blob/7d5c21cf1bd725a4264f53c990e5cc1b9da20985/src/page/welcome.rs#L11

HeitorAugustoLN replied:
Reported in rust-secure-code/cargo-auditable#225

HeitorAugustoLN committed:
doc/release-notes: add COSMIC Beta

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 440950`
Commit: `64325a4db47a4a1e99838e1eaa8b9564a82971c4`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 38 packages built:</summary>
  <ul>
    <li>cosmic-applets</li>
    <li>cosmic-applibrary</li>
    <li>cosmic-bg</li>
    <li>cosmic-comp</li>
    <li>cosmic-comp.debug</li>
    <li>cosmic-design-demo</li>
    <li>cosmic-edit</li>
    <li>cosmic-ext-calculator</li>
    <li>cosmic-ext-tweaks</li>
    <li>cosmic-files</li>
    <li>cosmic-greeter</li>
    <li>cosmic-icons</li>
    <li>cosmic-idle</li>
    <li>cosmic-initial-setup</li>
    <li>cosmic-launcher</li>
    <li>cosmic-notifications</li>
    <li>cosmic-osd</li>
    <li>cosmic-panel</li>
    <li>cosmic-player</li>
    <li>cosmic-randr</li>
    <li>cosmic-reader</li>
    <li>cosmic-screenshot</li>
    <li>cosmic-session</li>
    <li>cosmic-settings</li>
    <li>cosmic-settings-daemon</li>
    <li>cosmic-store</li>
    <li>cosmic-term</li>
    <li>cosmic-wallpapers</li>
    <li>cosmic-workspaces-epoch</li>
    <li>cosmic-workspaces-epoch.debug</li>
    <li>examine</li>
    <li>forecast</li>
    <li>libcosmicAppHook</li>
    <li>oboete</li>
    <li>quick-webapps</li>
    <li>tasks</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>

thefossguy commented:
We have a new tag: `epoch-1.0.0-beta.1.1`. Using this will match what's going to be in the Pop! _OS ISO.

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 440950`
Commit: `64325a4db47a4a1e99838e1eaa8b9564a82971c4`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 38 packages built:</summary>
  <ul>
    <li>cosmic-applets</li>
    <li>cosmic-applibrary</li>
    <li>cosmic-bg</li>
    <li>cosmic-comp</li>
    <li>cosmic-comp.debug</li>
    <li>cosmic-design-demo</li>
    <li>cosmic-edit</li>
    <li>cosmic-ext-calculator</li>
    <li>cosmic-ext-tweaks</li>
    <li>cosmic-files</li>
    <li>cosmic-greeter</li>
    <li>cosmic-icons</li>
    <li>cosmic-idle</li>
    <li>cosmic-initial-setup</li>
    <li>cosmic-launcher</li>
    <li>cosmic-notifications</li>
    <li>cosmic-osd</li>
    <li>cosmic-panel</li>
    <li>cosmic-player</li>
    <li>cosmic-randr</li>
    <li>cosmic-reader</li>
    <li>cosmic-screenshot</li>
    <li>cosmic-session</li>
    <li>cosmic-settings</li>
    <li>cosmic-settings-daemon</li>
    <li>cosmic-store</li>
    <li>cosmic-term</li>
    <li>cosmic-wallpapers</li>
    <li>cosmic-workspaces-epoch</li>
    <li>cosmic-workspaces-epoch.debug</li>
    <li>examine</li>
    <li>forecast</li>
    <li>libcosmicAppHook</li>
    <li>oboete</li>
    <li>quick-webapps</li>
    <li>tasks</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 38 packages built:</summary>
  <ul>
    <li>cosmic-applets</li>
    <li>cosmic-applibrary</li>
    <li>cosmic-bg</li>
    <li>cosmic-comp</li>
    <li>cosmic-comp.debug</li>
    <li>cosmic-design-demo</li>
    <li>cosmic-edit</li>
    <li>cosmic-ext-calculator</li>
    <li>cosmic-ext-tweaks</li>
    <li>cosmic-files</li>
    <li>cosmic-greeter</li>
    <li>cosmic-icons</li>
    <li>cosmic-idle</li>
    <li>cosmic-initial-setup</li>
    <li>cosmic-launcher</li>
    <li>cosmic-notifications</li>
    <li>cosmic-osd</li>
    <li>cosmic-panel</li>
    <li>cosmic-player</li>
    <li>cosmic-randr</li>
    <li>cosmic-reader</li>
    <li>cosmic-screenshot</li>
    <li>cosmic-session</li>
    <li>cosmic-settings</li>
    <li>cosmic-settings-daemon</li>
    <li>cosmic-store</li>
    <li>cosmic-term</li>
    <li>cosmic-wallpapers</li>
    <li>cosmic-workspaces-epoch</li>
    <li>cosmic-workspaces-epoch.debug</li>
    <li>examine</li>
    <li>forecast</li>
    <li>libcosmicAppHook</li>
    <li>oboete</li>
    <li>quick-webapps</li>
    <li>tasks</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cosmic-wallpapers</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cosmic-wallpapers</li>
  </ul>
</details>

HeitorAugustoLN committed:
cosmic-applets: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-applets/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-applibrary: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-applibrary/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-bg: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-bg/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-comp: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-comp/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-edit: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-edit/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-files: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-files/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-greeter: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-greeter/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-icons: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-icons/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-idle: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-idle/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-launcher: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-launcher/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-notifications: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-notifications/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-osd: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-osd/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-panel: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-panel/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-player: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-player/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-randr: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-randr/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-screenshot: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-screenshot/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-session: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-session/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-settings-daemon: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-settings-daemon/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-settings: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-settings/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-store: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-store/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-term: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-term/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-wallpapers: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-wallpapers/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

HeitorAugustoLN committed:
cosmic-workspaces-epoch: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/cosmic-workspaces-epoch/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

RossComputerGuy commented:
> https://system76.com/cosmic/download still points to https://github.com/lilyinstarlight/nixos-cosmic. Does someone know how to get that updated?

I have a few connections, I'll see if I can poke at this. I assume we'd want this pointing to nixpkgs since we have the packages & modules.

HeitorAugustoLN committed:
pop-launcher: 1.2.4 -> 1.2.6

Diff: https://github.com/pop-os/launcher/compare/1.2.4...1.2.6

HeitorAugustoLN committed:
pop-launcher: adopt by NixOS/cosmic

HeitorAugustoLN committed:
pop-launcher: remove `with lib` from meta

HeitorAugustoLN committed:
pop-launcher: migrate to `finalAttrs`

HeitorAugustoLN committed:
pop-launcher: `rev` -> `tag`

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 440950`
Commit: `55040b6e71b73ba446415b62671f18270a2a039b`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 40 packages built:</summary>
  <ul>
    <li>cosmic-applets</li>
    <li>cosmic-applibrary</li>
    <li>cosmic-bg</li>
    <li>cosmic-comp</li>
    <li>cosmic-comp.debug</li>
    <li>cosmic-design-demo</li>
    <li>cosmic-edit</li>
    <li>cosmic-ext-calculator</li>
    <li>cosmic-ext-tweaks</li>
    <li>cosmic-files</li>
    <li>cosmic-greeter</li>
    <li>cosmic-icons</li>
    <li>cosmic-idle</li>
    <li>cosmic-initial-setup</li>
    <li>cosmic-launcher</li>
    <li>cosmic-notifications</li>
    <li>cosmic-osd</li>
    <li>cosmic-panel</li>
    <li>cosmic-player</li>
    <li>cosmic-randr</li>
    <li>cosmic-reader</li>
    <li>cosmic-screenshot</li>
    <li>cosmic-session</li>
    <li>cosmic-settings</li>
    <li>cosmic-settings-daemon</li>
    <li>cosmic-store</li>
    <li>cosmic-term</li>
    <li>cosmic-wallpapers</li>
    <li>cosmic-workspaces-epoch</li>
    <li>cosmic-workspaces-epoch.debug</li>
    <li>examine</li>
    <li>forecast</li>
    <li>libcosmicAppHook</li>
    <li>oboete</li>
    <li>onagre</li>
    <li>pop-launcher</li>
    <li>quick-webapps</li>
    <li>tasks</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 40 packages built:</summary>
  <ul>
    <li>cosmic-applets</li>
    <li>cosmic-applibrary</li>
    <li>cosmic-bg</li>
    <li>cosmic-comp</li>
    <li>cosmic-comp.debug</li>
    <li>cosmic-design-demo</li>
    <li>cosmic-edit</li>
    <li>cosmic-ext-calculator</li>
    <li>cosmic-ext-tweaks</li>
    <li>cosmic-files</li>
    <li>cosmic-greeter</li>
    <li>cosmic-icons</li>
    <li>cosmic-idle</li>
    <li>cosmic-initial-setup</li>
    <li>cosmic-launcher</li>
    <li>cosmic-notifications</li>
    <li>cosmic-osd</li>
    <li>cosmic-panel</li>
    <li>cosmic-player</li>
    <li>cosmic-randr</li>
    <li>cosmic-reader</li>
    <li>cosmic-screenshot</li>
    <li>cosmic-session</li>
    <li>cosmic-settings</li>
    <li>cosmic-settings-daemon</li>
    <li>cosmic-store</li>
    <li>cosmic-term</li>
    <li>cosmic-wallpapers</li>
    <li>cosmic-workspaces-epoch</li>
    <li>cosmic-workspaces-epoch.debug</li>
    <li>examine</li>
    <li>forecast</li>
    <li>libcosmicAppHook</li>
    <li>oboete</li>
    <li>onagre</li>
    <li>pop-launcher</li>
    <li>quick-webapps</li>
    <li>tasks</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cosmic-wallpapers</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cosmic-wallpapers</li>
  </ul>
</details>

HeitorAugustoLN committed:
xdg-desktop-portal-cosmic: 1.0.0-beta.1 -> 1.0.0-beta.1.1

Diff: https://github.com/pop-os/xdg-desktop-portal-cosmic/compare/epoch-1.0.0-beta.1...epoch-1.0.0-beta.1.1

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 440950`
Commit: `ef354bbc2e1628d55d2bf417cefeae96923aea34`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 40 packages built:</summary>
  <ul>
    <li>cosmic-applets</li>
    <li>cosmic-applibrary</li>
    <li>cosmic-bg</li>
    <li>cosmic-comp</li>
    <li>cosmic-comp.debug</li>
    <li>cosmic-design-demo</li>
    <li>cosmic-edit</li>
    <li>cosmic-ext-calculator</li>
    <li>cosmic-ext-tweaks</li>
    <li>cosmic-files</li>
    <li>cosmic-greeter</li>
    <li>cosmic-icons</li>
    <li>cosmic-idle</li>
    <li>cosmic-initial-setup</li>
    <li>cosmic-launcher</li>
    <li>cosmic-notifications</li>
    <li>cosmic-osd</li>
    <li>cosmic-panel</li>
    <li>cosmic-player</li>
    <li>cosmic-randr</li>
    <li>cosmic-reader</li>
    <li>cosmic-screenshot</li>
    <li>cosmic-session</li>
    <li>cosmic-settings</li>
    <li>cosmic-settings-daemon</li>
    <li>cosmic-store</li>
    <li>cosmic-term</li>
    <li>cosmic-wallpapers</li>
    <li>cosmic-workspaces-epoch</li>
    <li>cosmic-workspaces-epoch.debug</li>
    <li>examine</li>
    <li>forecast</li>
    <li>libcosmicAppHook</li>
    <li>oboete</li>
    <li>onagre</li>
    <li>pop-launcher</li>
    <li>quick-webapps</li>
    <li>tasks</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 40 packages built:</summary>
  <ul>
    <li>cosmic-applets</li>
    <li>cosmic-applibrary</li>
    <li>cosmic-bg</li>
    <li>cosmic-comp</li>
    <li>cosmic-comp.debug</li>
    <li>cosmic-design-demo</li>
    <li>cosmic-edit</li>
    <li>cosmic-ext-calculator</li>
    <li>cosmic-ext-tweaks</li>
    <li>cosmic-files</li>
    <li>cosmic-greeter</li>
    <li>cosmic-icons</li>
    <li>cosmic-idle</li>
    <li>cosmic-initial-setup</li>
    <li>cosmic-launcher</li>
    <li>cosmic-notifications</li>
    <li>cosmic-osd</li>
    <li>cosmic-panel</li>
    <li>cosmic-player</li>
    <li>cosmic-randr</li>
    <li>cosmic-reader</li>
    <li>cosmic-screenshot</li>
    <li>cosmic-session</li>
    <li>cosmic-settings</li>
    <li>cosmic-settings-daemon</li>
    <li>cosmic-store</li>
    <li>cosmic-term</li>
    <li>cosmic-wallpapers</li>
    <li>cosmic-workspaces-epoch</li>
    <li>cosmic-workspaces-epoch.debug</li>
    <li>examine</li>
    <li>forecast</li>
    <li>libcosmicAppHook</li>
    <li>oboete</li>
    <li>onagre</li>
    <li>pop-launcher</li>
    <li>quick-webapps</li>
    <li>tasks</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cosmic-wallpapers</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cosmic-wallpapers</li>
  </ul>
</details>

GaetanLepage commented on pkgs/by-name/co/cosmic-applets/package.nix:
```diff
@@ -20,38 +20,32 @@
     dbus
     glib
     libinput
     pulseaudio
+    pipewire
```
Nit: keep the list sorted.

GaetanLepage reviewed (approved):
Went through the whole diff. Nothing to complain about. Thanks @HeitorAugustoLN!

RossComputerGuy commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 440950`
Commit: `55040b6e71b73ba446415b62671f18270a2a039b`

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 40 packages built:</summary>
  <ul>
    <li>cosmic-applets</li>
    <li>cosmic-applibrary</li>
    <li>cosmic-bg</li>
    <li>cosmic-comp</li>
    <li>cosmic-comp.debug</li>
    <li>cosmic-design-demo</li>
    <li>cosmic-edit</li>
    <li>cosmic-ext-calculator</li>
    <li>cosmic-ext-tweaks</li>
    <li>cosmic-files</li>
    <li>cosmic-greeter</li>
    <li>cosmic-icons</li>
    <li>cosmic-idle</li>
    <li>cosmic-initial-setup</li>
    <li>cosmic-launcher</li>
    <li>cosmic-notifications</li>
    <li>cosmic-osd</li>
    <li>cosmic-panel</li>
    <li>cosmic-player</li>
    <li>cosmic-randr</li>
    <li>cosmic-reader</li>
    <li>cosmic-screenshot</li>
    <li>cosmic-session</li>
    <li>cosmic-settings</li>
    <li>cosmic-settings-daemon</li>
    <li>cosmic-store</li>
    <li>cosmic-term</li>
    <li>cosmic-wallpapers</li>
    <li>cosmic-workspaces-epoch</li>
    <li>cosmic-workspaces-epoch.debug</li>
    <li>examine</li>
    <li>forecast</li>
    <li>libcosmicAppHook</li>
    <li>oboete</li>
    <li>onagre</li>
    <li>pop-launcher</li>
    <li>quick-webapps</li>
    <li>tasks</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>

RossComputerGuy reviewed (approved):
Approved automatically following the successful run of `nixpkgs-review`.

Thatoo commented:
Is it possible to backport it to stable 25.05?

ninelore commented:
fedora has backported the beta to 42 if that helps with deciding

Pheoxy commented:
@Thatoo @Pandapip1 @ninelore 

Given the scope of this change, it’s unlikely to be backported since stable branches generally only accept fixes rather than large new features. COSMIC Beta should be available in 25.11, which isn’t far off.

If you want to try it sooner switch to unstable or run it in a VM.
